pkg/networkmanager/test-utils.js

```diff
@@ -0,0 +1,275 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2014 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var utils = require("./utils");
+var cockpit = require("cockpit");
+var QUnit = require("qunit-tests");
+var assert = QUnit;
+
+function assert_throws(func, checks) {
+    var i;
+
+    assert.expect(checks.length);
+
+    function doit(i) {
```

Comment:
nitpick: this might be more readable and a bit more compact with forEach instead of passing an index, since we don't care about the actual index at all...

This actually applies to several functions in this pull request: it could help get rid of the extra loop variable `i` and structure the code a bit more obviously.